docs/config-v3-documentation.md

```diff
@@ -0,0 +1,953 @@
+
+### validator
+<!-- md:version 1.6.0 -->
+<!-- md:type run_ref -->
+<!-- md:default-empty -->
+
+Reference to the run section specifying how to run the validator.
+See [`[run]`](#run) for more.
+
+No value means no checking.
+
+### validator_type
+<!-- md:version 2.0.0 -->
+<!-- md:type enum -->
+<!-- md:flag required-applicable -->
+
+Specifies the validator type (required for non-empty [`validator`](#validator))
+
+- `simple-42` (recommended)
+- `simple-0`
+
+For more see [validator](task-parts/validator.md).
+
+### out_check
+<!-- md:version 1.0.0 -->
+<!-- md:type enum -->
+<!-- md:flag required -->
+
+Describes how to check outputs.
+
+- `diff` — compare with correct output (discouraged, can be slow in some cases)
+- `tokens` — compare token-by-token with correct output (tokens are separated by whitespace)
+- `shuffle` — like tokens, but allow permutation of tokens
+- `judge` — check with a custom program (called a 'judge')
+
+For [`task_type=interactive`](#task_type), only judge is allowed.
+
+For more see [batch checker](./task-parts/batch-checker.md) or [interactive judge](./task-parts/interactive-judge.md).
+
+### out_judge
+<!-- md:version 1.0.0 -->
+<!-- md:type run_ref -->
+<!-- md:flag required-applicable -->
+
+Only for [`out_check=judge`](#out_check) (required in that case).
+
+Reference to the run section specifying how to run the judge.
+See [`[run]`](#run) for more.
+
+### judge_type
+<!-- md:version 1.1.0 -->
+<!-- md:type enum -->
+<!-- md:flag required-applicable -->
+
+Only for [`out_check=judge`](#out_check) (required in that case).
+
+Specifies how to call the judge and how judge reports the verdict.
+
+For [`task_type=batch`](#task_type):
+
+- `cms-batch`
+- `opendata-v1`
+
+For [`task_type=interactive`](#task_type):
+
+- `cms-communication`
+
+See [batch checker](./task-parts/batch-checker.md) or [interactive judge](./task-parts/interactive-judge.md) for details.
+
+### judge_needs_in
+<!-- md:version 1.0.0 -->
+<!-- md:type bool -->
+<!-- md:default true -->
+
+Whether the judge needs the input for checking.
+
+### judge_needs_out
+<!-- md:version 1.0.0 -->
+<!-- md:type bool -->
+<!-- md:default true -->
+
+Whether the judge needs the correct output for checking.
+
+### tokens_ignore_newlines
+<!-- md:version 1.2.0 -->
+<!-- md:type bool -->
+<!-- md:default false -->
+
+Only for [`out_check=tokens`](#out_check).
+
+If set to `true`, newline characters will be ignored when checking the output,
+as if they were any other whitespace characters.
+Otherwise, newline characters are only ignored at the end of the file.
+
+### tokens_ignore_case
+<!-- md:version 1.2.0 -->
+<!-- md:type bool -->
+<!-- md:default false -->
+
+Only for [`out_check=tokens`](#out_check).
+
+If set to `true`, ASCII characters will be compared in a case-insensitive manner.
+
+### tokens_float_rel_error
+<!-- md:version 1.2.0 -->
+<!-- md:type float -->
+<!-- md:default-empty -->
+
```

Comment:
Again, I'd prefer not to have an empty description.

Is it possible to have one section for two keys?